cli/cmd/asset_create.go

```diff
@@ -0,0 +1,143 @@
+		if err != nil {
+			logger.Fatalf("Unable to get asset json file name: %v", err)
+		}
+		server, err := cmd.Flags().GetString("server")
+		if err != nil {
+			logger.Fatalf("Unable to get VMClarity server address: %v", err)
+		}
+
+		assetType, err := getAssetFromJSONFile(filename)
+		if err != nil {
+			logger.Fatalf("Failed to get asset from json file: %v", err)
+		}
+
+		_, err = assetType.ValueByDiscriminator()
+		if err != nil {
+			logger.Fatalf("Failed to determine asset type: %v", err)
+		}
+
+		assetID, err := createAsset(context.TODO(), assetType, server)
+		if err != nil {
+			logger.Fatalf("Failed to create asset: %v", err)
+		}
+		fmt.Println(assetID)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(assetCreateCmd)
+
+	assetCreateCmd.Flags().String("from-json-file", "", "asset json filename")
+	assetCreateCmd.Flags().String("server", "", "VMClarity server to create asset to, for example: http://localhost:9999/api")
+	if err := assetCreateCmd.MarkFlagRequired("from-json-file"); err != nil {
+		logger.Fatalf("Failed to mark from-json-file flag as required: %v", err)
+	}
+	if err := assetCreateCmd.MarkFlagRequired("server"); err != nil {
+		logger.Fatalf("Failed to mark server flag as required: %v", err)
+	}
+}
+
+func getAssetFromJSONFile(filename string) (*models.AssetType, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %v", err)
+	}
+	defer file.Close()
+
+	// get the file size
+	stat, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file stat: %v", err)
+	}
+	// read the file
+	bs := make([]byte, stat.Size())
+	_, err = file.Read(bs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file: %v", err)
+	}
+
+	assetType := &models.AssetType{}
+	if err := assetType.UnmarshalJSON(bs); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal asset into AssetType %v", err)
+	}
+
+	return assetType, nil
+}
+
+func createAsset(ctx context.Context, assetType *models.AssetType, server string) (string, error) {
+	client, err := backendclient.Create(server)
+	if err != nil {
+		return "", fmt.Errorf("failed to create VMClarity API client: %w", err)
+	}
+
+	creationTime := time.Now()
+	assetData := models.Asset{
+		AssetInfo: assetType,
+		LastSeen:  &creationTime,
+		FirstSeen: &creationTime,
+	}
+	asset, err := client.PostAsset(ctx, assetData)
+	if err == nil {
+		return *asset.Id, nil
+	}
+	var conflictError backendclient.AssetConflictError
+	if !errors.As(err, &conflictError) {
+		// If there is an error, and it's not a conflict telling
+		// us that the asset already exists, then we need to
+		// keep track of it and log it as a failure to
+		// complete discovery. We don't fail instantly here
+		// because discovering the assets is a heavy operation,
+		// so we want to give the best chance to create all the
+		// assets in the DB before failing.
+		return "", fmt.Errorf("failed to post asset: %v", err)
```

Comment:
I think this was a copy paste comment from the asset discovery code, we should update it.

I think we should also add a new flag to this command, `--update-if-exists` or something like that. If set then we will ignore the conflict error and do the patch. If not set then we should fail as with any other error.